akka-http/src/main/scala/akka/http/scaladsl/server/Rejection.scala

```diff
@@ -111,10 +110,39 @@ final case class InvalidOriginRejection(allowedOrigins: immutable.Seq[SHttpOrigi
  * Rejection created by unmarshallers.
  * Signals that the request was rejected because the requests content-type is unsupported.
  */
-final case class UnsupportedRequestContentTypeRejection(supported: immutable.Set[ContentTypeRange])
-  extends jserver.UnsupportedRequestContentTypeRejection with Rejection {
+final case class UnsupportedRequestContentTypeRejection(
+  supported:   Set[ContentTypeRange],
+  contentType: Option[ContentType]   = None)
+  extends jserver.UnsupportedRequestContentTypeRejection
+  with Rejection {
+
   override def getSupported: java.util.Set[model.ContentTypeRange] =
     scala.collection.mutable.Set(supported.map(_.asJava).toVector: _*).asJava // TODO optimise
+
+  @deprecated("for binary compatibility")
```

Comment:
All these should have a version as well, so `@deprecated("Something something", since = "10.1.6")`, they should also rather mention what to use instead. So, in this case "Use the two parameter constructor instead".

For the copy I guess the alternative now is to create a new object manually, so same there.